src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -90,7 +90,7 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
 
         private async Task DelayUntilFunctionDispatcherInitializedOrShutdown()
         {
-            if (_functionDispatcher != null && _functionDispatcher.State == FunctionInvocationDispatcherState.Initializing)
+            if (_functionDispatcher != null && _functionDispatcher.State != FunctionInvocationDispatcherState.Initialized)
             {
```

Comment:
as discussed offline, please make sure there is no delay in cases where function dispatcher never gets initialized.

Reply:
Especially as a hotfix, i think this is the safest change (to look at Initialized and Default). Let me know!